django/middleware/csrf.py

```diff
@@ -196,6 +227,25 @@ def _set_token(self, request, response):
             # Set the Vary header since content varies with the CSRF cookie.
             patch_vary_headers(response, ('Cookie',))
 
+    def _origin_verified(self, request):
+        request_origin = request.META['HTTP_ORIGIN']
+        good_origin = '%s://%s' % (
+            'https' if request.is_secure() else 'http',
+            request.get_host(),
+        )
+        if request_origin == good_origin:
+            return True
+        if request_origin in self.allowed_origins_exact:
```

Comment:
I had suggested doing this before computing `good_origin`: https://github.com/django/django/pull/13829#discussion_r579863426
That way you can the two method calls and string construction in favor of a set membership check.

Reply:
Oops, I changed it back yesterday because I think `request_origin == good_origin` (not overriding `CSRF_TRUSTED_ORIGINS`) is the more common case.

Reply:
Okay, whatever you decide should be good. Maybe you can include a code comment saying the reason for the order so that info won't be lost to future readers.

Reply:
I don't think there's a such strong argument either way that it needs a comment.